packages/vscode-extension/src/commands/CompareFile.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { UNRESTRICTED_FOLDERS } from '../ApplicationConstants';
+import { COMPARE_FILE, IMPORT_FILES } from '../service/TranslationKeys';
+import { actionResultStatus, ApplicationConstants } from '../util/ExtensionUtil';
+import FileImportCommon from './FileImportCommon';
+
+export default class CompareFile extends FileImportCommon {
+	private static readonly COMMAND_NAME = 'comparefile';
+    private static readonly TEMP_FOLDER_PREFIX = 'suitecloud-vscode-extension-compare-file-';
+
+	constructor() {
+		super(CompareFile.COMMAND_NAME);
+	}
+
+	protected validateBeforeExecute() {
+		const superValidation = super.validateBeforeExecute();
+		if (!superValidation.valid) {
+			return superValidation;
+		}
+
+		if (!this.activeFileIsUnderUnrestrictedFolder()) {
+			return this.unsuccessfulValidation(this.translationService.getMessage(IMPORT_FILES.ERROR.NOT_ALLOWED_FOLDER));
+		}
+
+		return this.successfulValidation();
+	}
+
+	protected async execute() {
+		const activeFilePath = this.activeFile!;
+		const tempFolderPath = fs.mkdtempSync(path.join(os.tmpdir(), CompareFile.TEMP_FOLDER_PREFIX));
+
+		this.copyManifestFileToTempFolder(tempFolderPath);
```

Comment:
why do we need to copy this?

Reply:
There are validations in the manifest file that is being performed before the action is done. So manifest file is also copied to the temporary directory. The temporary directory becomes a "clone" of the project. That's where the import files is actually being performed.

Reply:
Hey Touraj, the problem is that the cli (and sdf) needs to import the file to specific location, if not there is a validation error. 
To do so, we would have to change how file:import cli command is working, which is not part of the story (which only applies to vscode), Moreover, I don't know if we want this to work this way and give all users this option (even if it's hidden)

Anyways, I also see some possible problems with current solution (what to do if the command stops in the middle of the execution and the file is not copied back to the original place) So this it's a possible solution that we may want to discuss.